torch/_dynamo/guards.py

```diff
@@ -101,13 +101,38 @@ def sort_key(self):
     def __lt__(self, other):
         return self.sort_key() < other.sort_key()
 
+    @staticmethod
+    def weakref_to_str(obj_weakref):
+        """
+        This is a workaround of a Python weakref bug.
```

Comment:
Opened https://github.com/python/cpython/issues/99184 to track this bug at cpython.

Reply:
The cpython fix is ready: https://github.com/python/cpython/pull/99244